be/src/olap/storage_engine.cpp

```diff
@@ -1155,10 +1155,15 @@ Status StorageEngine::execute_task(EngineTask* task) {
 }
 
 // check whether any unused rowsets's id equal to rowset_id
-bool StorageEngine::check_rowset_id_in_unused_rowsets(const RowsetId& rowset_id) {
+std::optional<RowsetSharedPtr> StorageEngine::get_rowset_in_unused_rowsets(
```

Comment:
I think its very bad to get a rowset from gc space. GC is a standalone service. Other logic should not depend on it. The rowset is already hold in query context, I think you should get it from query context.

Reply:
done